# HiC data analysis

## HiC data processed with Juicer
- _inter_30.hic, .hic files for HiC contacts at MAPQ >= 30. [cite](https://arimagenomics.com/wp-content/files/Bioinformatics-User-Guide-Arima-HiC-and-Arima-High-Coverage-HiC.pdf)
- _inter_30.txt, the statistics files for Hi-C contacts at MAPQ >= 30. These are also stored within the respective .hic files in the header.
- _merged_loops.txt, generated by HiCCUPS inside of Juicer [file format](https://github.com/aidenlab/juicer/wiki/HiCCUPS#loop-list-content) [HiCCUPS](https://github.com/aidenlab/juicer/wiki/HiCCUPS)
- _contact_domains.txt, generated by HiCCUPS inside of Juicer 

## .hic file format
[.hic file format](https://github.com/aidenlab/hic-format/blob/master/HiCFormatV9.md)

## Workflow of Juicer (from fastq to .hic)
1. create an opt folder with 4 subdirs: [folder structure](https://bcm.app.box.com/v/juicerawsmirror/folder/11284128669)
- references
- work
- scripts
- restriction sites

2. Follow the installation instruction here (https://github.com/aidenlab/juicer/wiki/Installation)

3. Create restriction sites file (https://github.com/aidenlab/juicer/wiki/Installation)
- juicer/misc/generate_site_positions.py
- python generate_site_positions.py <enzyme> <genome ID>
  
4. Modify juicer.sh for ERISone
- juicer/LSF/scripts/juicer.sh

5. Tips
  - remove aligned folder before rerun to avoid error
  - need to copy scripts folder to /data/bioinformatics/tools/juicer/juicer  
  - need to copy scripts inside of juicer/LSF/scripts/ to /data/bioinformatics/tools/juicer/scripts (https://www.jianshu.com/p/4784c120b512)
  - add Arima) ligation to juicer.sh (https://github.com/aidenlab/juicer/issues/234)
 
  
## What does Salsa do?
  "The main advantage of Hi-C over previous methods is the ability to capture interactions over much larger genomic distances thereby producing scaffolds which can span a complete arm of the chromosome." [Scaffolding of long read assemblies using long range contact information, Jay Ghurye et.al, 2017]
  

## To generate .hic  from fastq
```
cd /data/bioinformatics/tools/juicer/work/HIC003
/data/bioinformatics/tools/juicer/scripts/juicer.sh -d /data/bioinformatics/tools/juicer/work/HIC003 \
  -p /data/bioinformatics/tools/juicer/restriction_sites/hg38.chrom.sizes \
  -y /data/bioinformatics/tools/juicer/restriction_sites/hg38_Arima.txt \
  -z /data/bioinformatics/referenceGenome/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/genome.fa \
  -D /data/bioinformatics/tools/juicer/juicer >juicer.log 2>&1
```

## Juicebox to visualize .hic file 
https://aidenlab.gitbook.io/juicebox/juicebox-web
  
  
# HiCDC+ workflow
  ## Finding Significant Interactions
  - construct_features, output is "hg38_50kb_GATC_GANTC_bintolen.txt.gz
  - generate_bintolen_gi_list, generate gi_list instance
  - add_hic_counts, add .hic counts
  - expand_1D_features, expand features for modeling
  - HiCDCPlus, this function finds significant interactions in a HiC-DC readable matrix and expresses statistical significance of counts through the following with a parallel implementation (using sockets; compatible with Windows): 'pvalue': significance P-value, 'qvalue': FDR corrected P-value, mu': expected counts, 'sdev': modeled standard deviation of expected counts.
  - result, 21118-03-01_inter_30_combined_result.hic, write normalized counts (observed/expected) to a .hic file
  - result, 21118-03-01_inter_30_combined_result.txt, write normalized counts (observed/expected) to a .txt file
  - output，a gi_list object

  ```
  > head(as.data.frame(gi_list[[1]][gi_list[[1]]$qvalue<=0.05]))
  seqnames1  start1    end1 width1 strand1    gc1  len1 seqnames2  start2    end2 width2 strand2
1      chr1 2400000 2450000  50001       * 0.5959 49538      chr1 3400000 3450000  50001       *
2      chr1 3000000 3050000  50001       * 0.5368 49894      chr1 3400000 3450000  50001       *
3      chr1 3900000 3950000  50001       * 0.5272 48448      chr1 5900000 5950000  50001       *
4      chr1 7950000 8000000  50001       * 0.4730 49698      chr1 8300000 8350000  50001       *
5      chr1 8000000 8050000  50001       * 0.4372 49539      chr1 8300000 8350000  50001       *
6      chr1 9250000 9300000  50001       * 0.5184 48766      chr1 9450000 9500000  50001       *
     gc2  len2       D counts    gc      len     mu   sdev         pvalue     qvalue
1 0.5584 49369 1000000     40 3.090  0.04231  4.598  3.926 0.000008243074 0.01045541
2 0.5584 49369  400000    115 2.570  0.09496 13.562 10.378 0.000002637755 0.00426998
3 0.4705 49480 2000000     28 1.625 -0.10474  2.506  2.391 0.000001171610 0.00241764
4 0.5019 49866  350000    157 1.407  0.13966 17.302 13.059 0.000000767771 0.00181313
5 0.5019 49866  300000    243 1.014  0.11610 20.887 15.626 0.000000007039 0.00004896
6 0.4953 49317  200000    284 1.797 -0.08090 32.160 23.697 0.000000820020 0.00187329
```
  ## Differential analysis using modified DESeq2
  - hicdcdiff
  - output, DESeq-processed matrices (in a .txt file)
```
(base) [tz949@eris1n3 diff_analysis_example]$ zcat diff_resACMoverSR_chr10.txt.gz | head
chr	startI	startJ	baseMean	log2FoldChange	lfcSE	stat	pvalue	padj
chr10	3050000	4750000	13.4484170978077	0.220460389571675	0.304025046536862	0.725138905767574	0.46836681848552	0.999290292241651
chr10	3100000	4800000	13.8056106336287	-0.0796136036770952	0.316773434663169	-0.25132664221591	0.801561578553612	0.999290292241651
chr10	3150000	4350000	20.3762058756698	0.268680127030688	0.270998185361081	0.99144622194682	0.32146773969325	0.999290292241651
chr10	3150000	4600000	23.468292142183	0.278903447170748	0.261286430864027	1.06742415305864	0.285780346586615	0.999290292241651
chr10	3150000	4650000	34.0331078251787	0.0721756736041125	0.229352538346074	0.314693153712585	0.752994642899563	0.999290292241651
chr10	3150000	4700000	33.3009402269902	-0.00525310243161406	0.232534331515613	-0.0225906531623756	0.981976799622494	0.999290292241651
chr10	3150000	4750000	73.7449331664501	0.0156950151676941	0.199510810233499	0.0786674924999067	0.937297102547809	0.999290292241651
chr10	3150000	4800000	27.6565439780342	0.236774331200241	0.242904634632286	0.974762509404875	0.329678077828567	0.999290292241651
chr10	3200000	3750000	76.0127604814415	0.183274509750465	0.199625873942567	0.918089955629666	0.358571772975857	0.999290292241651
```
  ## ICE normalization using HiTC
ICE (iterative correction and eigenvector decomposition) normalization (Imakaev et al. 2012a) functions by modeling the expected 𝐼𝐼𝐹𝐹𝑖𝑖𝑖𝑖 for every pair of regions (i,j) as Eij=BiBjTij, where 𝐵𝐵𝑖𝑖 and 𝐵𝐵𝑖𝑖 are the biases and 𝑇𝑇𝑖𝑖𝑖𝑖 is the true matrix of normalized IFs. The maximum likelihood solution for the biases 𝐵𝐵𝑖𝑖 is obtained by iterative correction. It attempts to make all regions equally visible and was shown to perform as well as the explicit bias correction method by Yaffe and Tanay (Belton et al. 2012). ICE normalization can be performed using the HiTC R package’s normICE function or as a step of the Hi-C processing pipeline HiC-Pro.
- hic2icenorm_gi_list
- inout file is .hic from Juicer
- output a ICE normalized .hic file to the input .hic path
```
> head(gi_list[[sample]])
$chr1
GInteractions object with 333178 interactions and 2 metadata columns:
           seqnames1             ranges1     seqnames2             ranges2 |    counts         D
               <Rle>           <IRanges>         <Rle>           <IRanges> | <numeric> <integer>
       [1]      chr1       750000-800000 ---      chr1       750000-800000 | 3131.1830         0
       [2]      chr1       800000-850000 ---      chr1       750000-800000 |  170.6725     50000
       [3]      chr1       850000-900000 ---      chr1       750000-800000 |   36.5086    100000
       [4]      chr1       900000-950000 ---      chr1       750000-800000 |   15.0402    150000
       [5]      chr1      950000-1000000 ---      chr1       750000-800000 |   15.7209    200000
       ...       ...                 ... ...       ...                 ... .       ...       ...
  [333174]      chr1 248700000-248750000 ---      chr1 248900000-248950000 |   42.8217    200000
  [333175]      chr1 248750000-248800000 ---      chr1 248900000-248950000 |   45.1133    150000
  [333176]      chr1 248800000-248850000 ---      chr1 248900000-248950000 |  223.3130    100000
  [333177]      chr1 248850000-248900000 ---      chr1 248900000-248950000 |  424.2610     50000
  [333178]      chr1 248900000-248950000 ---      chr1 248900000-248950000 | 2498.6877         0
  -------
  regions: 4394 ranges and 0 metadata columns
  seqinfo: 1 sequence from an unspecified genome; no seqlengths
```
                                                                
## Finding TADs using TopDom
HiCDCPlus converts the gi_list instance with ICE normalized counts into TAD annotations through an implementation of TopDom. 
https://github.com/HenrikBengtsson/TopDom
https://github.com/jasminezhoulab/TopDom/blob/master/TopDom%20Manual_v0.0.2.pdf
- gi_list_topdom
```
$chr6
$chr6$binSignal
     id  chr from.coord to.coord local.ext mean.cf        pvalue
1     1 chr6     150000   200000         0  140.95 0.70922949878
2     2 chr6     200000   250000         0  133.18 0.99890500028
3     3 chr6     250000   300000         0  114.81 0.99972475233
4     4 chr6     300000   350000         0   93.49 0.98759726449
5     5 chr6     350000   400000         0   71.41 0.61237715408
6     6 chr6     400000   450000         0   56.77 0.05156772206
7     7 chr6     450000   500000         0   47.91 0.00108598591
8     8 chr6     500000   550000        -1   39.43 0.00000027796
9     9 chr6     550000   600000         0   41.03 0.00036761144
10   10 chr6     600000   650000         1   53.89 0.83880180345                 
```
                 
```
$chr6$domain
     chr from.id from.coord to.id to.coord      tag    size
1   chr6       1     150000     8   550000   domain  400000
2   chr6       9     550000    13   800000   domain  250000
3   chr6      14     800000    29  1600000   domain  800000
4   chr6      30    1600000    39  2100000   domain  500000
5   chr6      40    2100000    52  2750000   domain  650000
6   chr6      53    2750000    59  3100000   domain  350000
7   chr6      60    3100000    62  3250000   domain  150000
8   chr6      63    3250000    71  3700000   domain  450000
9   chr6      72    3700000    78  4050000   domain  350000
10  chr6      79    4050000    81  4200000 boundary  150000
```
                                                                
```
$chr6$bed
    chrom chromStart  chromEnd     name
1    chr6     150000    550000   domain
2    chr6     550000    800000   domain
3    chr6     800000   1600000   domain
4    chr6    1600000   2100000   domain
5    chr6    2100000   2750000   domain
6    chr6    2750000   3100000   domain
7    chr6    3100000   3250000   domain
8    chr6    3250000   3700000   domain
9    chr6    3700000   4050000   domain
10   chr6    4050000   4200000 boundary
```
 ### How does TopDom define TAD?
(https://watermark.silverchair.com/gkv1505.pdf?token=AQECAHi208BE49Ooan9kkhW_Ercy7Dm3ZL_9Cf3qfKAc485ysgAAAuAwggLcBgkqhkiG9w0BBwagggLNMIICyQIBADCCAsIGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMTiszqrgbqboDyn0dAgEQgIICk78wVC3y3coO-YPdTKsVvob4iXU8TXZWzrudPeinr7_FCD-ExIB4JzICjAs6TZJ-AHM0F2Y4YY_TOmmihijtDyc1-n3-0RrftbgBLDV9YqDNpapRegrqIDfr3MwCQX_7GpG5I1RyHEeDMx0j8Sdoc5CkHsgN1JtXtMRrrY9H1IqoEVwHQKLYZXiUzxWVxlTCMO9EtFKbG8ej7e42tlgu1_dXtZg2xJ95aoaGY2ozBcFN4rZ9u3YednoUDfiF4tghOA0YT5dKprJVzh-MwCZMLrHScK5XdRxB5fXPuUMbQuHb8Ff1eafePawMim4R0wyq8X2X8p3huiC8XkG3PyxFDfUB0H3nozBFfSroaTrPSzW8y8Q088UC_SMk0FZu6qeRIuKxe2NIbkHC-zZJROhAEeFmfwdpwCfvLoL6SiMsp_a0XaBPvEKouHnHQfeiKII86Wntbmy4hpn779L3yiSu1cnYKpjETJalwT-qkOcljYHbOgz75DKUmxefDwQacoqPniEM_3B0UB3BxJgfVmpBnbuF4In7GfeDX1ddPISfHanp0LoDTrnKFXD99qoqH7VyWELtWuMGUUtv-GD0knm9pO4Ettlaxm4Q6qzE5RGtCjOosA3zqos9jW0tNpr228S_Xt8hekpqziTBOMVsJRF47Hp32vY1pLYGq0ts4cOZS-n7aqB4R58vyENE0TN9II5qIgw0pqwizyI1QXgTT7-tKxVLGxPdLCXER6LRCYU1drND7EKVm7MP0s9NsPzhjdZwOffzAkrO4ofycCpvxUGk9JDLXS3jr9ACtWiaM7qCPQ9TY7dMTcBxyH-0CsEMlk7Hxxj5YcQoFgSoy_0qlAbAras_n0dfiznKhmuvbk6OQc8XrTTY)                 
The input data are a Hi-C contact map, where entries are contact frequencies between any two chromatin segments (i.e. bins in the data matrix). Our method has three steps: (i) For each bin, we generate a value binSignal by computing the average contact frequency among pairs of chromatin regions (one upstream, the other downstream) in a small window surrounding the bin. This step results in a curve binSignal(i) that runs along the chromosome. (ii) Discover TD boundaries as local minima in the binSignal(i) series. (iii) Filter out false detections in the local minima by statistical testing. Each step is described in more detail below.
                 
## Finding A/B compartment using Juicer
HiCDCPlus can call Juicer eigenvector function to determine A/B compartments from .hic files.                                                                
                                                                
Regions that have more frequent interactions are classified as A compartment, representing transcription active compartment, regions with less interactions are classified as B compartment, representing transcription inactive compartment. 
https://github.com/Jfortin1/TCGA_AB_Compartments
                 https://link.springer.com/protocol/10.1007/978-1-4939-8766-5_16

### What are the diagnosis plots from hicdcdiff?
geomean_sizefactors_chr1.pdf
diagnostic plots of the normalization factors, geometric means of such factors by distance bin
