# HiC data analysis

## HiC data processed with Juicer
- _inter_30.hic, .hic files for HiC contacts at MAPQ >= 30. [cite](https://arimagenomics.com/wp-content/files/Bioinformatics-User-Guide-Arima-HiC-and-Arima-High-Coverage-HiC.pdf)
- _inter_30.txt, the statistics files for Hi-C contacts at MAPQ >= 30. These are also stored within the respective .hic files in the header.
- _merged_loops.txt, generated by HiCCUPS inside of Juicer [file format](https://github.com/aidenlab/juicer/wiki/HiCCUPS#loop-list-content) [HiCCUPS](https://github.com/aidenlab/juicer/wiki/HiCCUPSDiff)
- _contact_domains.txt, generated by HiCCUPS inside of Juicer 

## .hic file format
[.hic file format](https://github.com/aidenlab/hic-format/blob/master/HiCFormatV9.md)

## Workflow of Juicer (from fastq to .hic)
1. create an opt folder with 4 subdirs: [folder structure](https://bcm.app.box.com/v/juicerawsmirror/folder/11284128669)
- references
- work
- scripts
- restriction sites

2. Follow the installation instruction here (https://github.com/aidenlab/juicer/wiki/Installation)

3. Create restriction sites file (https://github.com/aidenlab/juicer/wiki/Installation)
- juicer/misc/generate_site_positions.py
- python generate_site_positions.py <enzyme> <genome ID>
  
4. Modify juicer.sh for ERISone
- juicer/LSF/scripts/juicer.sh

5. Tips
  - remove aligned folder before rerun to avoid error
  - need to copy scripts folder to /data/bioinformatics/tools/juicer/juicer  
  - need to copy scripts inside of juicer/LSF/scripts/ to /data/bioinformatics/tools/juicer/scripts (https://www.jianshu.com/p/4784c120b512)
  - add Arima) ligation to juicer.sh (https://github.com/aidenlab/juicer/issues/234)
 
  
## What does Salsa do?
  "The main advantage of Hi-C over previous methods is the ability to capture interactions over much larger genomic distances thereby producing scaffolds which can span a complete arm of the chromosome." [Scaffolding of long read assemblies using long range contact information, Jay Ghurye et.al, 2017]
  

## To generate .hic  from fastq
```
cd /data/bioinformatics/tools/juicer/work/HIC003
/data/bioinformatics/tools/juicer/scripts/juicer.sh -d /data/bioinformatics/tools/juicer/work/HIC003 \
  -p /data/bioinformatics/tools/juicer/restriction_sites/hg38.chrom.sizes \
  -y /data/bioinformatics/tools/juicer/restriction_sites/hg38_Arima.txt \
  -z /data/bioinformatics/referenceGenome/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/genome.fa \
  -D /data/bioinformatics/tools/juicer/juicer >juicer.log 2>&1
```

## Juicebox to visualize .hic file 
https://aidenlab.gitbook.io/juicebox/juicebox-web
  
  
